File: languages/python/bloock/entity/identity/integer_enum_attribute_descriptor.py
from __future__ import annotations
from typing import List

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.attribute_descriptor import AttributeDescriptor


class IntegerEnumAttributeDescriptor(AttributeDescriptor):
    """
    Represents a descriptor for an attribute with an integer enum value.
    """
    def __init__(self, display_name: str, technical_name: str, description: str, required: bool, enumeration: List[int]) -> None:
        """
        Constructs an IntegerEnumAttributeDescriptor object with the specified parameters.
        :type enumeration: object
        :type required: object
        :type description: object
        :type technical_name: object
        :type display_name: object
        :rtype: object
        """
        super().__init__(display_name, technical_name, description, required)
        self.enumeration = enumeration

    @staticmethod
    def from_proto(a: proto.IntegerEnumAttributeDefinition) -> IntegerEnumAttributeDescriptor:
        return IntegerEnumAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
            enumeration=list(a.enum),
        )

    def to_proto(self) -> proto.IntegerEnumAttributeDefinition:
        return proto.IntegerEnumAttributeDefinition(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
            enum=list(self.enumeration),
        )

File: languages/php/Bloock/Entity/Identity/IntegerEnumAttributeDescriptor.php
<?php

namespace Bloock\Entity\Identity;

use Bloock\IntegerEnumAttributeDefinition;
use Google\Protobuf\Internal\RepeatedField;

/**
 * Represents a descriptor for an attribute with an integer enum value.
 */
class IntegerEnumAttributeDescriptor extends AttributeDescriptor
{
    private RepeatedField $enumeration;

    /**
     * Constructs an IntegerEnumAttributeDescriptor object with the specified parameters.
     * @param string $displayName
     * @param string $technicalName
     * @param string|null $description
     * @param bool $required
     * @param RepeatedField $enumeration
     */
    public function __construct(string $displayName, string $technicalName, ?string $description, bool $required, RepeatedField $enumeration)
    {
        parent::__construct($displayName, $technicalName, $description, $required);
        $this->enumeration = $enumeration;
    }

    public static function fromProto(IntegerEnumAttributeDefinition $res): IntegerEnumAttributeDescriptor
    {
        return new IntegerEnumAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired(), $res->getEnum());
    }

    public function toProto(): IntegerEnumAttributeDefinition {
    
        $p = new IntegerEnumAttributeDefinition();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        $p->setEnum($this->enumeration);
        return $p;
    }
}